gnd/src/main/java/com/google/android/gnd/persistence/sync/LocalMutationSyncWorker.java

```diff
@@ -110,10 +117,32 @@ private Completable processMutations(ImmutableList<Mutation> mutations, String u
   private Completable processMutations(ImmutableList<Mutation> mutations, User user) {
     return remoteDataStore
         .applyMutations(mutations, user)
+        .andThen(processPhotoFieldMutations(mutations))
         // TODO: If the remote sync fails, reset the state to DEFAULT.
         .andThen(localDataStore.finalizePendingMutations(mutations));
   }
 
+  /**
+   * Filter all mutations containing observation mutations with changes to photo fields and uploads
+   * to remote storage.
+   */
+  private Completable processPhotoFieldMutations(ImmutableList<Mutation> mutations) {
+    return Observable.fromIterable(mutations)
+        .filter(mutation -> mutation instanceof ObservationMutation)
+        .cast(ObservationMutation.class)
+        .flatMapCompletable(
+            mutation ->
+                Observable.fromIterable(mutation.getResponseDeltas())
+                    .filter(delta -> delta.getFieldType() == Type.PHOTO)
+                    .map(ResponseDelta::getNewResponse)
+                    .map(Optional::isPresent)
+                    .map(Object::toString)
+                    .flatMapCompletable(
```

Comment:
I think this can be simplified, but not sure of the exact syntax.

Reply:
Done a478804